src/main/java/seedu/address/logic/parser/task/TaskSearchCommandParser.java

```diff
@@ -76,6 +95,82 @@ public TaskSearchCommand parse(String args) throws ParseException {
+                    }
+                } else { // mm/YY
+                    if (isOutOfBoundsMonth(keywords.get("month"))) {
+                        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                                TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                    }
+                }
+            } else { // dd/MM, painful combination to check
+
+                if (isOutOfBoundsMonth(keywords.get("month"))) {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                }
+
+                if (keywords.get("month") == FEBRUARY) {
+                    if (!(MINIMUM_DATE <= keywords.get("date")
+                            && (keywords.get("date")) <= TWENTY_NINE_MAXIMUM_DATE)) { // not valid days in Feb
+                        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                                TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                    }
+                } else if (THIRTY_DAY_MONTHS.contains(keywords.get("month"))) { // thirty day months
+                    if (!(MINIMUM_DATE <= keywords.get("date")
+                            && (keywords.get("date") <= THIRTY_MAXIMUM_DATE))) { // not valid days in Feb
+                        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                                TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                    }
+                } else { // thirty one day months
+                    if (isOutOfBoundsDate(keywords.get("date"))) {
+                        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                                TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                    }
+                }
+            }
+        } else if (oneParameterProvided) {
+            if (keywords.containsKey("date")) {
+                if (isOutOfBoundsDate(keywords.get("date"))) {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                }
+            } else if (keywords.containsKey("month")) {
+                if (isOutOfBoundsMonth(keywords.get("month"))) {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                }
+            } else if (keywords.containsKey("year")) {
+                if (isOutOfBoundsYear(keywords.get("year"))) {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            TaskSearchCommand.MESSAGE_OUT_OF_BOUNDS_VALUES));
+                }
+            }
+        }
+
+        System.out.println(keywords);
```

Comment:
Should abstract with methods
Too many nested if-else blocks -> hard to debug + low readability

Reply:
Can merge, I will try to fix that later. I couldn't think of a better way for the logic as of now, since that is just the rule of how the date works. I have abstracted it out using `final` constants.

Reply:
The `^` in `has[0] ^ check[0]` just checks if `has[0] != check[0]`

Reply:
Thanks, fixing it right now!